kotest-assertions/kotest-assertions-core/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/AssertSoftlyTest.kt

```diff
@@ -154,6 +155,21 @@ class AssertSoftlyTest : FreeSpec({
          }
       }
 
+
+      "Should not loose stacktrace with only one assertion" {
+         shouldThrow<AssertionError> {
+            assertSoftly {
+               null should beEmpty()
+            }
+         }.run {
+            message.shouldContainInOrder(
+               "The following assertion failed:",
+               "1) Expecting actual not to be null",
+               "   at com.sksamuel.kotest.matchers.AssertSoftlyTest${'$'}1${'$'}1${'$'}9.invokeSuspend(AssertSoftlyTest.kt:162)"
```

Comment:
Minor but blocking: let's remove line number 162 from expected string number 3, to make the test less fragile. Otherwise any refactoring/reformatting will break this test.

Reply:
That makes sense. However, this was one of the problems reported in the issue: an incorrect pointer in the stack trace. I'll move that test to a separate class to prevent it from affecting the other tests. What do you think?

Reply:
> That makes sense. However, this was one of the problems reported in the issue: an incorrect pointer in the stack trace. I'll move that test to a separate class to prevent it from affecting the other tests. What do you think?

I concur - let's extract this test into its own file.

Reply:
Please take a look now, I changed the hardcoded line number to dynamically calculated variable.

Reply:
> Please take a look now, I changed the hardcoded line number to dynamically calculated variable.

LGTM, started the build

Reply:
> Please take a look now, I changed the hardcoded line number to dynamically calculated variable.

the following test is failing, and it might be related: "adds an Exception to an empty collection of assertion failures"

Reply:
Yes, I see that there is a test class marked to run on Linux only, which is why I didn't catch it failing in my local run.

I've already fixed it. The issue was that the expected exception class had changed, since any assertion failure in the `assertSoftly` block should now be classified as a `MultiAssertionError`, even if only one assertion fails.